### ISSUE TITLE
post_updates fails with large datasets

### ISSUE THREAD

**Issue by tompropst** [attm2x/m2x-python#25]
`post_updates` is failing when attempting to add fairly large data sets. I assume there is a practical limit that we can adhere to but that should be documented or added as parameter we could check if not already available (I could not find it).

**Comment by kristinpeterson** [comment]
@tompropst thank you for reporting! Could you share the following when you have a moment?

* an example of the dataset you are trying to post
* error output received when `post_updates` fails

**Comment by tompropst** [comment]
@kristinpeterson , sorry for the slow response. We weren't seeing the error until we created a test case without exception handling. Now we see:

````
raise HTTPError(http_error_msg, response=self)
requests.exceptions.HTTPError: 400 Client Error: Bad Request for url: https://api-m2x.att.com/v2/devices/<my_device_key_was_here>/updates
````

We are just uploading data that looks like the following:

````
{'timestamp': '2016-12-31T04:07:39.851514Z', 'value': '0.28\n'}
````

The issue is that at times we are posting a few thousand data points (6800 for example).

**Comment by kristinpeterson** [comment]
@tompropst the `400 Bad Request` error is returned from the API whenever the data being sent is in a format that is not expected by the api [1].

The `post_updates` method is making a call to the Post Device Updates (Multiple Values to Multiple Streams) endpoint[2] expects the data to be formatted as follows:

```
{ "values": {
    "temperature": [
      { "timestamp": "2014-09-09T19:15:00.981Z", "value": 32 },
      { "timestamp": "2014-09-09T20:15:00.124Z", "value": 30 },
      { "timestamp": "2014-09-09T21:15:00.124Z", "value": 15 } ],
    "humidity": [
      { "timestamp": "2014-09-09T19:15:00.874Z", "value": 88 },
      { "timestamp": "2014-09-09T20:15:00.752Z", "value": 60 },
      { "timestamp": "2014-09-09T21:15:00.752Z", "value": 75 } ]
  }
}
```

Can you model your data per the above example and try again? Also, I notice that there's a newline (`/n`) character at the end of the value, might be worthwhile to also remove that in case that is causing any issues as `value` expects just a float value.

[1]: https://m2x.att.com/developer/documentation/v2/overview#API-Error-Codes
[2]: https://m2x.att.com/developer/documentation/v2/device#Post-Device-Updates--Multiple-Values-to-Multiple-Streams-

**Comment by tompropst** [comment]
I believe the format of the data is correct. The snippet I provided was just an example of a single timestamp / value pair. We can successfully upload such data (with the rest of the structure you show) until we have a backlog of data greater than 1000 data points. This sometimes happens if we lose network connectivity or some other, intermittent error occurs.

Now that we know there is a 1000 data point limit on a single post request, we have put a check in place to divide large data sets before posting. I opened this issue because I couldn't find this limit documented. It can be closed now unless you want to keep it open as a feature request for the documentation.

Thanks for your help.
